server/src/main/java/io/kroki/server/service/KrokiBlockedThreadChecker.java

```diff
@@ -0,0 +1,85 @@
+package io.kroki.server.service;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxException;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.impl.VertxInternal;
+import io.vertx.core.impl.btc.BlockedThreadEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class KrokiBlockedThreadChecker {
+
+  private static final Logger logger = LoggerFactory.getLogger(KrokiBlockedThreadChecker.class);
+
+  private final int evenLoopPoolSize;
+  private final int workerPoolSize;
+  private final Duration trackStatsFor;
+
+  private final Map<String, Instant> eventLoopStats = new ConcurrentHashMap<>();
+  private final Map<String, Instant> workerStats = new ConcurrentHashMap<>();
```

Comment:
I'm a bit concerned since we don't do eviction since is basically a memory leak. We could use something like https://github.com/ben-manes/caffeine to use a time-based eviction.

Reply:
Sure, I switched to Caffeine's Cache to store the stats. 

I was reluctant to use Caffeine as it was not included in the project and I didn't expect the 2 maps to grow above 100 entries.